syntax="proto2";

//类似于C++中的命名空间namespace
package doc_index_proto;

message Pair
{
    required int32 begin = 1;
    required int32 end = 2;
};

//正排索引
message DocInfo
{
    //数组下标（文档id）
    required uint64 doc_id = 1;

    //文档内容
    required string title = 2;
    required string content = 3;
    //URL
    required string jump_url = 4;
    required string show_url = 5;
    //分词结果
    repeated Pair title_token = 6;
    repeated Pair content_token = 7;
};

//文档id列表
message Weight
{
    //文档id
    required uint64 doc_id = 1;
    //权重
    required int32 weight = 2;
    //关键词出现次数
    required int32 title_count = 3;
    required int32 content_count = 4;
    //关键词第一次在文档中出现的位置
    required int32 first_pos = 5;
};

//倒排索引
message KwdInfo
{
    //关键词
    required string key = 1;
    //文档id列表
    repeated Weight doc_list = 2;
};

//类似于C++中的类class
message Index
{
    //正排索引
    repeated DocInfo forward_index = 1;
    //倒排索引
    repeated KwdInfo inverted_index = 2;
};

